document publicly curated modules

rberlind opened the pull request:
This PR documents the addition and use of publicly curated modules in the Terraform Cloud private module registry. It adds information about publicly curated modules to the PMR index page and also adds a new "Adding Public Modules" page to the PMR group supplementing the "Publishing Private Modules" page. It also makes some other changes to existing pages in the PMR group to reflect the facts that there are now 2 kinds of modules in a PMR (public and private) and the fact that the menu has changed from "Modules" to "Registry".

I've updated many but not all of the images that had the old "Modules" menu.  I could not do the ones related to shared modules in TFE since I do not have access to a TFE server at this time.

laurapacilio commented on content/source/docs/cloud/registry/index.html.md:
```diff
@@ -2,13 +2,21 @@
+
 > **Hands-on:** Try the [Share Modules in the Private Module Registry](https://learn.hashicorp.com/tutorials/terraform/module-private-registry?in=terraform/modules&utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) tutorial on HashiCorp Learn.
 
 Terraform Cloud's private module registry helps you share [Terraform modules](/docs/language/modules/index.html) across your organization. It includes support for module versioning, a searchable and filterable list of available modules, and a configuration designer to help you build new workspaces faster.
 
 By design, the private module registry works much like the [public Terraform Registry](/docs/registry/index.html). If you're already used the public registry, Terraform Cloud's registry will feel familiar.
+
+The private module registry lets you add both private and public modules to the registry, so that you can more easily find all your modules in one place. Private modules are modules that you publish directly to the registry. Public modules, also referred to as "publicly curated modules", live in the public Terraform Registry and are synchronized from it. By adding some of the many public Terrafor ...
+
+-> **Note:** Publicly curated modules are not supported in Terraform Enterprise.
+
+You can even use a [Sentinel Policy](/docs/cloud/sentinel/index.html) to mandate that all non-root modules used in Terraform configurations be private or publicly curated modules from your own private module registry. You can even require that all those modules use recent versions with a policy like [use-recent-versions-from-pmr.sentinel](https://github.com/hashicorp/terraform-guides/blob/master/ ...
```
Thank you so much for adding this - it's great!! I'm just reorganizing it a tiny bit and adding some links to other pages. See what you think!

```suggestion
Terraform Cloud's private module registry works similarly to the [public Terraform Registry](/docs/registry/index.html) and helps you share [Terraform modules](/docs/language/modules/index.html) across your organization. It includes support for module versioning, a searchable and filterable list of available modules, and a configuration designer to help you build new workspaces faster.  

You can add both private and public modules to the registry:

- [Private modules](/docs/cloud/registry/publish.html) directly to the registry are hosted on the registry and are only available to members of that organization. In Terraform Enterprise, they are also available to organizations that are configured to [share modules](/docs/enterprise/admin/module-sharing.html) with that organization. 
- [Public modules](/docs/cloud/registry/add.html), are automatically synchronized from the Terraform Registry where they are hosted. Public modules are not supported in Terraform Enterprise.

You can use [Sentinel Policies](/docs/cloud/sentinel/index.html) to enforce rules about how members of your organization can use these modules. For example, you can mandate that all non-root modules in Terraform configurations be private or public modules from your own private module registry. You can even require that all of those modules use recent versions with a policy like [use-recent-versions-from-pmr.sentinel](https://github.com/hashicorp/terraform-guides/blob/master/governance/third-generation/cloud-agnostic/http-examples/use-recent-versions-from-pmr.sentinel).
```

rberlind replied:
I'm fine with the changes, but think you need some slight changes onl your new lines 18 and 19.  Line 18 seems to need a verb like "are added" after the link.  And line 19 should remove the comma after the link.

laurapacilio commented on content/source/docs/cloud/registry/publish.html.md:
```diff
@@ -60,15 +60,15 @@ that don't look like version numbers are ignored.
+To publish a module, navigate to the modules list with the "Registry" button and click the "Publish" button in the upper right and then the "Module" button that appears under it.
 
-![Terraform Cloud screenshot: the "modules" button and the "+Add Module" button](./images/publish-add-button.png)
+![Terraform Cloud screenshot: the "registry" button and the "+Add Module" button](./images/publish-add-button.png)
```
This picture doesn't have Registry in it - I think it should be in the main navigation bar, right?

rberlind replied:
Yes, it is under the Registry menu.  I just pushed new image that has that.

laurapacilio commented on content/source/docs/cloud/registry/using.html.md:
```diff
@@ -17,16 +17,17 @@ A Terraform Cloud private module registry has a few key differences from the [pu
+All users in your organization can view your private module registry. To find available modules, click the "Registry" button in the Terraform Cloud main navigation bar.
 
-![Terraform Cloud screenshot: Navigation bar with modules button highlighted](./images/using-modules-button.png)
+![Terraform Cloud screenshot: Navigation bar with registry button highlighted](./images/using-modules-button.png)
```
This image also still has "Modules" in the menu bar. Should it?

rberlind replied:
I did not update all images.  Only the ones I could.  I don't have TFE server to update images related to shared organizations.  Originally, I was only going to update index page and add my new add public modules page, but then decided to improve other pages too.  I figured making corrections was better than leaving them the way they were even without updating images.  Perhaps you can update the other images?

laurapacilio commented on content/source/docs/cloud/registry/using.html.md:
```diff
@@ -65,7 +66,9 @@ Use the "Examples" dropdown to navigate to example modules and use the  Readme/I
-## Using Private Modules in Configurations
+## Using Modules in Configurations
+
+The syntax for referencing publicly curated modules in the [module block](/docs/language/modules/syntax.html) `source` argument is `<NAMESPACE>/<MODULE NAME>/<PROVIDER>`.
```
Could we add an example underneath this one like there is for the private modules? I think that would help users!

rberlind replied:
Yes.  It would be something like `hashicorp/subnets/cidr`

laurapacilio reviewed (changes requested):
Hey @rberlind Really really great work on this! As this is a pretty large change, I had a few suggestions to streamline the content and also a couple of questions/suggestions that I thought you could take a look at. Let me know what you think of these, and then once we've addressed them I'd love to take another look through. 

Thank you so much for adding this and especially for thinking about all of the pages that this information should live on! :-)

rberlind commented on content/source/docs/cloud/registry/add.html.md:
```diff
@@ -0,0 +1,58 @@
+
+[permissions-citation]: #intentionally-unused---keep-for-maintainers
+
+-> **API:** See the [Registry Modules API](../api/modules.html#create-a-module-with-no-vcs-connection-). If you use the API to add a public module, be sure to create a module with no VCS connection and to set the `registry_name` property to `public`.
```
I think the information already is on the API page, so it is fine to delete this.

rberlind commented on content/source/docs/cloud/registry/add.html.md:
```diff
@@ -0,0 +1,58 @@
+
+This brings you to the "Search Public Modules" page, which has a text field that you can use to search for public modules in the public Terraform registry.
+
+![Terraform Cloud screenshot: the "search public modules" page, with a provider and module name entered](./images/add-search-public-modules.png)
```
I was just imitating what we already had for working with private modules.

rberlind commented:
I accepted almost all your changes, made tiny tweaks to one, updated one image, added example of use of public module and explained why I did not update some images to use "Registry" in menu intead of "Modules", namely that I don't have access to a TFE server currently.  Perhaps you can update the remaining images?

laurapacilio reviewed (approved):
A couple of final nits to polish the text - please see what you think!  

Also...did you fix those two images? I can't find the ones now that had Modules instead of Registry in this PR anymore! However, I do see that there are several other screenshots on these pages that still have "Modules" at the top of the screenshots where I think it should be "Registry." I can open a separate PR to deal with that issue, so don't let that block you from merging this work.  Does that sound okay?
